### PR TITLE
Add Traversable instance to LocalValue

### LINKED ISSUE #42: add Traversable instance for LocalValue
I want to be able to write, for instance:

``` purescript
now <- nowDateTime
let now' = (traverse (adjust $ Minutes 3.0) now)
```

where `now'` is a `Maybe LocalDateTime`.

Using the functor instance, e.g., `map (adjust $ Minutes 3.0) now` gives me a `LocalValue (Maybe DateTime)` which isn't so useful.

### PR DESCRIPTION
**Pull request by anilanar**
Fixes #42.

Let me know if there's room for improvement (I'm sure there is).

@garyb Thanks for pointing me to this issue, I had to think about what `Traversable` and `Foldable` exactly mean.

### CODE REVIEW

File: test/Test/Main.purs [hunk 1]
```diff
@@ -1,24 +1,25 @@
 module Test.Main where
 
 import Prelude
-
```
Comment (anilanar) [inline test/Test/Main.purs]: Hmm, the vim psc-ide auto-sorted imports. I'll revert this.

File: test/Test/Main.purs [hunk 2]
```diff
@@ -142,6 +143,28 @@ main = do
   assert $ Instant.toDateTime (Instant.fromDateTime dt3) == dt3
   assert $ Instant.toDateTime (Instant.fromDateTime dt4) == dt4
 
+  -- locale ------------------------------------------------------------------
+
+  let locale = Locale.Locale (Just $ Locale.LocaleName "UTC")
+               $ Duration.Minutes 0.0
+  let crLocalVal = Locale.LocalValue locale
+
```
Comment (anilanar) [inline test/Test/Main.purs]: I'm not sure if these assertions are useful at all or how I can improve them. There aren't many edge cases with an applicative that can hold a single element.

**Comment by anilanar** [comment]
This is ready for review from my side.

**Comment by garyb** [comment]
Looks good, thanks!
